aequilibrae/project/network/link_types.py

```diff
@@ -0,0 +1,129 @@
+
+
+class LinkTypes:
+    """
+    Access to the API resources to manipulate the link_types table in the network
+
+    ::
+
+        from aequilibrae import Project
+
+        p = Project()
+        p.open('path/to/project/folder')
+
+        link_types = p.network.link_types
+
+        # We can get a dictionary of all modes in the model
+        all_link_types = modes.all_types()
+
+        #And do a bulk change and save it
+        for link_type_id, link_type_obj in all_link_types.items():
+            link_type_obj.beta = 1
+
+        # We can save changes for all link types in one go
+        all_link_types.save()
+
+        # or just get one link_type in specific
+        default_link_type = link_types.get('y')
+
+        # or just get it by name
+        default_link_type = link_types.get_by_name('default')
+
+        # We can change the description of the mode
+        default_link_type.description = 'My own new description'
+
+        # Let's say we are using alpha to store lane capacity during the night as 90% of the standard
+        default_link_type.alpha =0.9 * default_link_type.lane_capacity
+
+        # To save this mode we can simply
+        default_link_type.save()
+
+        # We can also create a completely new link_type and add to the model
+        new_type = new('a')
+        new_type.link_type_name = 'Arterial'  # Only ASCII letters and *_* allowed
+        # other fields are not mandatory
+
+        # We then save it to the database
+        new_type.save()
+
+        # we can even keep editing and save it directly once we have added it to the project
+        new_type.lanes = 3
+        new_type.lane_capacity = 1100
+        new_type.save()
+
+    """
+    __items = {}
+
+    def __init__(self, net):
+        self.__all_types = []
+        self.conn = net.conn  # type: Connection
+        self.curr = net.conn.cursor()
+
+        tl = TableLoader()
+        link_types_list = tl.load_table(self.curr, 'link_types')
+        existing_list = [lt['link_type_id'] for lt in link_types_list]
+        if link_types_list:
+            self.__properties = list(link_types_list[0].keys())
+        for lt in link_types_list:
+            if lt['link_type_id'] not in self.__items:
+                self.__items[lt['link_type_id']] = LinkType(lt)
+
+        to_del = [key for key in self.__items.keys() if key not in existing_list]
+        for key in to_del:
+            del self.__items[key]
+
+    def new(self, link_type_id: str) -> LinkType:
+        if link_type_id in self.__items:
+            raise ValueError('Link Type ID already exists in the model. It must be unique.')
+
+        tp = {key: None for key in self.__properties}
+        tp['link_type_id'] = link_type_id
+        lt = LinkType(tp)
+        self.__items[link_type_id] = lt
+        logger.warning('Link type has not yet been saved to the database. Do so explicitly')
+        return lt
+
+    def drop(self, link_type_id: str) -> None:
```

Comment:
this should probably be delete - drop is a term generally used to refer to deleting a table/view/db entity

Reply:
Agreed